x-pack/plugins/code/server/security.ts

```diff
@@ -61,14 +61,14 @@ export class SecureRoute {
     }
 
     this.server.securedRoute = securedRoute;
-  }
+}
 
   private isSecurityEnabledInEs() {
     const xpackInfo = this.server.plugins.xpack_main.info;
-    if (
-      xpackInfo.isAvailable() &&
-      (!xpackInfo.feature('security').isEnabled() || xpackInfo.license.isOneOf('basic'))
-    ) {
+    if (!xpackInfo.isAvailable()) {
+      throw Boom.methodNotAllowed('x-pack info is not available yet.');
```

Comment:
I wouldn't recommend throwing a `405` in these situations. A `405` is generally used when the incorrect http verb is used with a URL. I'd recommend a [503](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503).